libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/http/DefaultHttpRequestComposer.kt

```diff
@@ -147,22 +149,33 @@ class DefaultHttpRequestComposer(
         operation: Operation<D>,
         customScalarAdapters: CustomScalarAdapters,
         sendApqExtensions: Boolean,
+        sendEnhancedClientAwarenessExtensions: Boolean,
         query: String?,
     ): Map<String, Upload> {
       return composePostParams(
-          writer, operation, customScalarAdapters, query, apqExtensionsWriter(operation.id(), sendApqExtensions)
+          writer, operation, customScalarAdapters, query, extensionsWriter(operation.id(), sendApqExtensions, sendEnhancedClientAwarenessExtensions)
       )
     }
 
-    private fun apqExtensionsWriter(id: String, sendApqExtensions: Boolean): JsonWriter.() -> Unit {
+    private fun extensionsWriter(apqId: String, sendApqExtensions: Boolean, sendEnhancedClientAwarenessExtensions: Boolean): JsonWriter.() -> Unit {
+      if (!sendApqExtensions && !sendEnhancedClientAwarenessExtensions) return { }
+
       return {
-        if (sendApqExtensions) {
-          name("extensions")
-          writeObject {
+        name("extensions")
+        writeObject {
+          if (sendApqExtensions) {
```

Comment:
Can you make it so that we don't send an empty object if there are neither apqs nor client awareness?

Reply:
Avoiding an empty object should be handled by the `if` above no? That's an early exit for neither of the extensions being set and by here we know we're sending an object so these `if` statements are to only include the sub-objects that we need.